test/e2e/kubectl/kubectl.go

```diff
@@ -1697,6 +1697,282 @@ metadata:
+			}
+			containers := rc.Spec.Template.Spec.Containers
+			if containers == nil || len(containers) != 1 || containers[0].Image != nginxImage {
+				framework.Failf("Failed creating rc %s for 1 pod with expected image %s", name, nginxImage)
+			}
+			framework.WaitForRCToStabilize(c, ns, name, framework.PodStartTimeout)
+
+			By("deleting the rc " + name)
+			framework.RunKubectlOrDie("delete", "rc", name, nsFlag, "--cascade=true")
+
+			By("verifying rc was deleted")
+			err = wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+				_, err := c.Core().ReplicationControllers(ns).Get(name, metav1.GetOptions{})
+				if apierrs.IsNotFound(err) {
+					return true, nil
+				} else if err != nil {
+					return false, err
+				}
+				return false, nil
+			})
+			if err != nil {
+				framework.Failf("Failed verifying rc was deleted: %v", err)
+			}
+
+			By("verifying pods were deleted")
+			label := labels.SelectorFromSet(labels.Set(map[string]string{"run": name}))
+			podList, err := c.Core().Pods(ns).List(metav1.ListOptions{LabelSelector: label.String()})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(podList.Items)).To(Equal(0))
+		})
+
+		It("should delete replicaset and all of the resources managed by it", func() {
+			name = "e2e-test-nginx-replicaset"
+			labelFlag = fmt.Sprintf("--labels=run=%v", name)
+
+			By("creating the replicaset using a YAML file")
+			framework.RunKubectlOrDie("create -f hack/testdata/frontend-replicaset.yaml", nsFlag, labelFlag)
+
+			By("verifying the pod controlled by " + name + " gets created")
+			label := labels.SelectorFromSet(labels.Set(map[string]string{"run": name}))
+			err := testutils.WaitForPodsWithLabelRunning(c, ns, label)
+			if err != nil {
+				framework.Failf("Failed waiting for pod controlled by %s to be running: %v", name, err)
+			}
+
+			By("deleting the replicaset " + name)
+			framework.RunKubectlOrDie("delete", "replicaset", name, nsFlag, "--cascade=true")
+
+			By("verifying replicaset was deleted")
+			err = wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+				_, err := c.ExtensionsV1beta1().ReplicaSets(ns).Get(name, metav1.GetOptions{})
+				if apierrs.IsNotFound(err) {
+					return true, nil
+				} else if err != nil {
+					return false, err
+				}
+				return false, nil
+			})
+			if err != nil {
+				framework.Failf("Failed verifying replicaset was deleted: %v", err)
+			}
+
+			By("verifying pods were deleted")
+			podList, err := c.Core().Pods(ns).List(metav1.ListOptions{LabelSelector: label.String()})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(podList.Items)).To(Equal(0))
+		})
+
+		It("should delete daemonset and all of the resources managed by it", func() {
+			name = "e2e-test-nginx-daemonset"
+			labelFlag = fmt.Sprintf("--labels=run=%v", name)
+
+			By("creating the daemonset using a YAML file")
+			framework.RunKubectlOrDie("create -f hack/testdata/rollingupdate-daemonset.yaml", nsFlag, labelFlag)
```

Comment:
@janetkuo `podAntiAffinity` was introduced in v1.6, but we plan to run this test against v1.5 cluster. i am not sure whether this YAML file is appropriate or not.